astroquery/esa/neocc/core.py

```diff
@@ -0,0 +1,453 @@
+        the list of all NEAs currently considered in the NEOCC system.
+
+        >>> from astroquery.esa.neocc import neocc
+        >>> list_data = neocc.query_list(list_name='nea_list')  # doctest: +REMOTE_DATA
+        >>> print(list_data)  # doctest: +IGNORE_OUTPUT +REMOTE_DATA
+             NEA
+        ---------------
+               433 Eros
+             719 Albert
+             887 Alinda
+           1036 Ganymed
+              1221 Amor
+            1566 Icarus
+                    ...
+                 2023RD
+                 2023RE
+                 2023RF
+                6344P-L
+        Length = 32558 rows
+
+        **Close approaches (upcoming):**  The output of this list is a `~astropy.table.Table` which
+        contains object with upcoming close approaches.
+
+        >>> from astroquery.esa.neocc import neocc
+        >>> list_data = neocc.query_list(list_name='close_approaches_upcoming')  # doctest: +REMOTE_DATA
+        >>> print(list_data)  # doctest: +IGNORE_OUTPUT +REMOTE_DATA
+        Object Name           Date          ... Rel. vel in km/s CAI index
+        ----------- ----------------------- ... ---------------- ---------
+            2021JA5 2023-09-07 00:00:00.000 ...             11.0     3.496
+            2023QC5 2023-09-08 00:00:00.000 ...              7.6     2.662
+             2020GE 2023-09-08 00:00:00.000 ...              1.4     3.308
+            2020RS1 2023-09-10 00:00:00.000 ...              9.1     4.071
+            2023QE8 2023-09-10 00:00:00.000 ...             14.5     1.039
+                ...                     ... ...              ...       ...
+           2012SX49 2024-08-29 00:00:00.000 ...              4.3     2.665
+           2016RJ20 2024-08-30 00:00:00.000 ...             14.8     2.118
+             2021JT 2024-09-02 00:00:00.000 ...              8.3     4.216
+           2021RB16 2024-09-02 00:00:00.000 ...              8.5     3.685
+            2007RX8 2024-09-02 00:00:00.000 ...              7.0     2.322
+        Length = 176 rows
+
+        Note
+        ----
+        If the contents request fails the following message will be printed:
+
+        *Initial attempt to obtain list failed. Reattempting...*
+
+        Then a second request will be automatically sent to the NEOCC portal.
+        """
+
+        # Get URL to obtain the data from NEOCC
+        url = lists.get_list_url(list_name)
+
+        # Request list two times if the first attempt fails
+        try:
+            # Parse decoded data
+            neocc_list = lists.get_list_data(url, list_name)
+
+            return neocc_list
+
+        except ConnectionError:  # pragma: no cover
+            print('Initial attempt to obtain list failed. Reattempting...')
+            # Wait 5 seconds
+            time.sleep(5)
+            # Parse decoded data
+            neocc_list = lists.get_list_data(url, list_name)
+
+            return neocc_list
+
+    @staticmethod
+    def query_object(name, tab, **kwargs):
```

Comment:
could you explicitely list the possible kwargs instead of hiding them in **kwargs? 

Doing that helps avoid user bugs, as well as making things a bit more explicit. Doing so, please add  a `*` in the signature, so all those kwargs are keyword only.

Reply:
Done